File: app/src/main/java/com/nagot/lastfm/ui/artistinfo/ArtistInfoFragmentMvpPresenter.java
package com.nagot.lastfm.ui.artistinfo;

import com.nagot.lastfm.base.BaseMvpPresenter;

/**
 * Created by IanNagot on 16/03/2018.
 */

public interface ArtistInfoFragmentMvpPresenter<V extends ArtistInfoFragmentMvpView> extends BaseMvpPresenter<V> {

    void getArtistInfo(String artistName);
}

File: app/src/main/java/com/nagot/lastfm/ui/artistinfo/ArtistInfoFragmentMvpPresenter.kt
package com.nagot.lastfm.ui.artistinfo

import com.nagot.lastfm.base.BaseMvpPresenter

/**
 * Created by Nagot on 18/03/2018.
 */
interface ArtistInfoFragmentMvpPresenter<V: ArtistInfoFragmentMvpView>: BaseMvpPresenter<V> {

    fun getArtistInfo(artistName: String)
}